nltk/text.py

```diff
@@ -530,16 +534,50 @@ def dispersion_plot(self, words):
         from nltk.draw import dispersion_plot
 
         dispersion_plot(self, words)
-
-    def generate(self, words):
-        """
-        Issues a reminder to users following the book online
-        """
-        import warnings
-
-        warnings.warn(
-            'The generate() method is no longer available.', DeprecationWarning
-        )
+        
+    def build_ngram_lm(self, tokenized_sents, n=3):
+        train_data, padded_sents = padded_everygram_pipeline(n, tokenized_sents)
+        model = MLE(order=n)
+        model.fit(train_data, padded_sents)
+        return model
+        
+    def generate(self, length=100, text_seed=None, random_seed=42):
+        """
+        Print random text, generated using a trigram language model.
+        See also `help(nltk.lm)`.
+        
+        :param length: The length of text to generate (default=100)
+        :type length: int
+        
+        :param text_seed: Generation can be conditioned on preceding context.
+        :type text_seed: list(str)
+        
+        :param random_seed: A random seed or an instance of `random.Random`. If provided,
+        makes the random sampling part of generation reproducible.
+        :type random_seed: int
+        
+        """
+        # Create the model when using it the first time.
+        self.tokenized_sents = [sent.split(' ') for sent in 
```

Comment:
Should we consider making this a private attribute? Seems a bit scary to me that by calling `generate` we change the public state of `Text` by either adding a new attribute or re-setting an existing one. Also note, we perform this operation even if the ngram model is already defined (i.e. we don't actually *need* the tokenized sentences then).

Reply:
The weird thing is Text don't keep sentence tokenization. And sentence tokenization splits on `'\n` which gutenberg (I think it's extracted from pdf) actually has sentences spanning across `\n`.

Reply:
What do you mean by "don't keep sentence tokenization"? The attribute is being set, so that's definitely kept after the generation. If it only gets reset the next time generation happens, why don't we use a function-scoped variable for it?

Reply:
I mean the original input to the Text object already lost all notion of sentence because it was using the `word_tokenize` directly (which flattens the sentences).

Reply:
Change it's to a private attribute convention `_tokenized_sents `.